src/rgw/rgw_op.cc

```diff
@@ -5933,6 +5933,19 @@ void RGWBulkUploadOp::execute()
     return;
   }
 
+  std::string bucket_path, file_prefix;
+  if (! s->init_state.url_bucket.empty()) {
+    file_prefix = bucket_path = s->init_state.url_bucket + "/";
+    if (!s->object.empty()) {
```

Comment:
OK, `rgw_obj_key::empty` checks whether its `name` is empty, so we can safely examine the last element of `object_name`.